ts/sagas/wallet.ts

```diff
@@ -24,12 +77,161 @@ function* fetchTransactions(
 }
 
 function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
+  loadCards: () => Promise<ReadonlyArray<Wallet>>
 ): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+  const cards: ReadonlyArray<Wallet> = yield call(loadCards);
   yield put(cardsFetched(cards));
 }
 
+const navigateTo = (routeName: string, params?: object) => {
+  return NavigationActions.navigate({ routeName, params });
+};
+
+function* paymentSaga(): Iterator<Effect> {
+  yield put(navigateTo(ROUTES.PAYMENT_SCAN_QR_CODE)); // start by showing qr code scanner
+
+  while (true) {
+    const action = yield take([
+      PAYMENT_INSERT_DATA_MANUALLY,
+      PAYMENT_SHOW_SUMMARY,
+      PAYMENT_CONFIRM_SUMMARY,
+      PAYMENT_PICK_PAYMENT_METHOD,
+      PAYMENT_CONFIRM_PAYMENT_METHOD,
+      PAYMENT_REQUEST_OTP,
+      PAYMENT_VERIFY_OTP,
+      COMPLETE_PAYMENT
```

Comment:
this is missing the request for paymentId to the app backend (that is async and could take some time or could fail)

Reply:
it is part of a PR that builds on top of this one -- will be publishing it asap